packages/atomic/src/components/commerce/atomic-commerce-search-box-recent-queries/atomic-commerce-search-box-recent-queries.spec.ts

```diff
@@ -0,0 +1,364 @@
+      const event = buildCustomEvent(
+        'atomic/searchBoxSuggestion/register',
+        vi.fn()
+      );
+      expect(dispatchSpy).toHaveBeenCalledWith(event);
+    });
+
+    it('should assign the bindings from the event', async () => {
+      expect(element.bindings).toBeDefined();
+      expect(element.bindings).toEqual(
+        expect.objectContaining({
+          ...defaultBindings,
+          i18n: expect.anything(),
+        })
+      );
+    });
+
+    it('should call initialize after assigning bindings', async () => {
+      expect(initializeSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('#initialize', () => {
+    it('should create a recent queries list controller', async () => {
+      const {element} = await renderElements();
+
+      expect(buildRecentQueriesList).toHaveBeenCalledWith(
+        element.bindings.engine,
+        {
+          initialState: {
+            queries: ['query1', 'query2', 'query3'],
+          },
+          options: {
+            clearFilters: element.bindings.clearFilters,
+            maxLength: 1000,
+          },
+        }
+      );
+    });
+
+    it('should subscribe to the recent queries list', async () => {
+      await renderElements();
+
+      expect(buildFakeRecentQueriesList().subscribe).toHaveBeenCalledWith(
+        expect.any(Function)
+      );
+    });
+
+    it('should return a position that is the index of the element', async () => {
+      const {element, searchBox} = await renderElements();
+
+      const object = element.initialize();
+
+      const position = Array.from(searchBox.children).indexOf(element);
+      expect(object.position).toBe(position);
+    });
+
+    describe('renderItems', () => {
+      describe('when analytics are disabled', () => {
+        it('should return an empty array', async () => {
+          vi.mocked(buildRecentQueriesList).mockReturnValue(
+            buildFakeRecentQueriesList({analyticsEnabled: false})
+          );
+          const {element} = await renderElements({
+            engine: {
+              logger: {
+                warn: vi.fn(),
+              },
+            },
+          });
+
+          const object = element.initialize();
+
+          expect(object.renderItems()).toEqual([]);
+        });
+      });
+
+      describe('when analytics are enabled', () => {
+        let element: AtomicCommerceSearchBoxRecentQueries;
+        let object: SearchBoxSuggestions;
+        let items: SearchBoxSuggestionElement[];
+
+        beforeEach(async () => {
+          vi.mocked(buildRecentQueriesList).mockReturnValue(
+            buildFakeRecentQueriesList({
+              queries: ['query1', 'query2', 'query3'],
+            })
+          );
+          ({element} = await renderElements());
+          object = element.initialize();
+          items = object.renderItems();
+        });
+
+        it('should return the correct number of items', async () => {
+          expect(items.length).toBe(4);
```

Comment:
It's four because the clear button is counted, right?

That's a bit counterintuitive, we should add a comment.